packages/framework/presence/src/test/latestValueManager.spec.ts

```diff
@@ -3,12 +3,206 @@
+
+			clock.setSystemTime(initialTime);
+
+			// Set up the presence connection
+			presence = prepareConnectedPresence(runtime, "sessionId-2", "client2", clock, logger);
+		});
+
+		afterEach(function (done: Mocha.Done) {
+			clock.reset();
+
+			// If the test passed so far, check final expectations.
+			if (this.currentTest?.state === "passed") {
+				assertFinalExpectations(runtime, logger);
+			}
+			done();
+		});
+
+		after(() => {
+			clock.restore();
+		});
+
+		it("sends update signal with updated data", async () => {
+			// Setup
+			runtime.signalsExpected.push(
+				[
+					"Pres:DatastoreUpdate",
+					{
+						"sendTimestamp": 1010,
+						"avgLatency": 10,
+						"data": {
+							"system:presence": {
+								"clientToSessionId": {
+									"client2": { "rev": 0, "timestamp": 1000, "value": "sessionId-2" },
+								},
+							},
+							"s:name:testStateWorkspace": {
+								"data": {
+									"sessionId-2": { "rev": 0, "timestamp": 1010, "value": { "num": 0 } },
+								},
+							},
+						},
+					},
+				],
+				[
+					"Pres:DatastoreUpdate",
+					{
+						"sendTimestamp": 1010,
+						"avgLatency": 10,
+						"isComplete": true,
+						"data": {
+							"system:presence": {
+								"clientToSessionId": {
+									"client2": { "rev": 0, "timestamp": 1000, "value": "sessionId-2" },
+								},
+							},
+							"s:name:testStateWorkspace": {
+								"data": {
+									"sessionId-2": { "rev": 0, "timestamp": 1010, "value": { "num": 0 } },
+								},
+							},
+						},
+					},
+				],
+				[
+					"Pres:DatastoreUpdate",
+					{
+						"sendTimestamp": 1010,
+						"avgLatency": 10,
+						"isComplete": true,
+						"data": {
+							"system:presence": {
+								"clientToSessionId": {
+									"client2": { "rev": 0, "timestamp": 1000, "value": "sessionId-2" },
+								},
+							},
+							"s:name:testStateWorkspace": {
+								"data": {
+									"sessionId-2": { "rev": 0, "timestamp": 1010, "value": { "num": 0 } },
+								},
+							},
+						},
+					},
+				],
+				[
+					"Pres:DatastoreUpdate",
+					{
+						"sendTimestamp": 1020,
+						"avgLatency": 10,
+						"data": {
+							"system:presence": {
+								"clientToSessionId": {
+									"client2": { "rev": 0, "timestamp": 1000, "value": "sessionId-2" },
+								},
+							},
+							"s:name:testStateWorkspace": {
+								"data": {
+									"sessionId-2": { "rev": 1, "timestamp": 1020, "value": { "num": 42 } },
+								},
+							},
+						},
+					},
+				],
+				[
+					"Pres:DatastoreUpdate",
+					{
+						"sendTimestamp": 1030,
+						"avgLatency": 10,
+						"data": {
+							"system:presence": {
+								"clientToSessionId": {
+									"client2": { "rev": 0, "timestamp": 1000, "value": "sessionId-2" },
+								},
+							},
+							"s:name:testStateWorkspace": {
+								"data": {
+									"sessionId-2": { "rev": 2, "timestamp": 1030, "value": { "num": 65 } },
+								},
+							},
+						},
+					},
+				],
```

Comment:
If we drop the two processSignal calls, then I think this is the only signal expected when batching is implemented. The sendTimestamp should be 1070. (60 past when first local update was set at 1010.)
The test will also need a clock advance after last local set. I think you can clock.tick(1000) large number and timer would still happen at the right time.